server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -766,17 +766,21 @@ private void preAllocatedBufferCheck() {
           removeIds.add(info.getRequireId());
         }
       }
+      int expiredBufferIdsCnt = 0;
       for (Long requireId : removeIds) {
         PreAllocatedBufferInfo info = requireBufferIds.remove(requireId);
         if (info != null) {
           // move release memory code down to here as the requiredBuffer could be consumed during
           // removing processing.
           shuffleBufferManager.releaseMemory(info.getRequireSize(), false, true);
+          ShuffleServerMetrics.counterExpiredPreAllocatedBufferSizeTotal.inc(info.getRequireSize());
```

Comment:
CounterExpiredPreAllocatedBufferSizeTotal is necessary? I feel A bit repetitive with counterExpiredPreAllocatedBufferIdTotal. 
If we're just trying to capture how often timeouts happen, counterExpiredPreAllocatedBufferIdTotal maybe enough. But I'm not sure if you serve any other purpose.

Reply:
the total expired buffer size is to diagnose cases where allocated metric is negative. I will compare these value to check what happen of allocated is leaked.